plugins/crd/plugin_impl_crd.go

```diff
@@ -243,51 +245,213 @@ func createCRD(plugin *Plugin, FullName, Group, Version, Plural, Name string) er
 }
 
 func informerNetworkServices(plugin *Plugin) {
-	plugin.informerNS = plugin.sharedFactory.Networkservice().V1().NetworkServices().Informer()
+	// We use a shared informer from the informer factory, to save calls to the
+	// API as we grow our application and so state is consistent between our
+	// control loops. We set a resync period of 30 seconds, in case any
+	// create/replace/update/delete operations are missed when watching
+	plugin.sharedFactoryNS = factory.NewSharedInformerFactory(plugin.crdClient, time.Second*30)
+
+	plugin.informerNS = plugin.sharedFactoryNS.Networkservice().V1().NetworkServices().Informer()
 	// We add a new event handler, watching for changes to API resources.
 	plugin.informerNS.AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc: networkserviceEnqueue,
+			AddFunc: func(obj interface{}) {
+				var newEvent handler.NsmEvent
+				var err error
+				newEvent.Key, err = cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+				newEvent.EventType = handler.HandlerCreate
+				newEvent.ResourceType = handler.NetworkServiceResource
+				if err == nil {
+					queueNS.Add(newEvent)
+				}
+			},
 			UpdateFunc: func(old, cur interface{}) {
 				if !reflect.DeepEqual(old, cur) {
-					networkserviceEnqueue(cur)
+					// For an update event, we delete the old and add the current
+					var newEventOld, newEventCur handler.NsmEvent
+					var err error
+					newEventOld.Key, err = cache.DeletionHandlingMetaNamespaceKeyFunc(old)
+					newEventOld.EventType = handler.HandlerDelete
```

Comment:
Curious if it could happen,  we enqueue `delete` of old object and `add` of new one, when `workforever` gets event of delete and then by some reason cannot complete it and put it back on the queue and picks up next in line which is `add` object. So we have old object, trying to add new object, it will not happen, as there is already an object with the same name, then when `delete` finally succeed, we end up without object at all. I understand that this scenario is kind of a corner case, but still worth to discuss.

Reply:
The only way we re-queue something is when GetByKey() returns an error. It's a definite corner case, and comes along as a problem when we're breaking an update operation down to "delete/create."

Reply:
If we get rid of worker queue then this scenario will not be possible.

Reply:
As discussed in IRC, lets leave worker queue for now, noting this could be an issue.